aten/src/TH/THRandom.c

```diff
@@ -200,11 +200,33 @@ uint64_t THRandom_random(THGenerator *_generator)
   return y;
 }
 
+uint64_t THRandom_random64(THGenerator *_generator)
+{
+  uint64_t hi = THRandom_random(_generator);
```

Comment:
add a comment / TODO here that THRandom_random is only generating 32-bit of randomness